sct-commons/src/main/java/org/lfenergy/compas/sct/commons/ExtRefEditorService.java

```diff
@@ -399,6 +349,65 @@ private String computeDaiValue(String lnPrefix, TExtRef extRef, String daName) {
+        }
+       // Analog
+        if (setting.getChannelType().equals(TChannelType.ANALOG)) {
+            lnEditor.findLn(tlDevice, tAnyLN -> LnId.from(tAnyLN).lnClass().equals(LN_RADR) && LnId.from(tAnyLN).lnInst().equals(setting.getChannelNum()))
+                    .ifPresent(tln -> updateDaiValue(dtt, tied, tlDevice, tln, tExtRef, setting));
+            lnEditor.findLn(tlDevice, tAnyLN -> LnId.from(tAnyLN).lnClass().equals(LN_RADR) && LnId.from(tAnyLN).lnInst().equals(setting.getChannelNum())
+                            && LnId.from(tAnyLN).prefix().equals(LN_PREFIX_A))
+                    .ifPresent(tln -> updateDaiValue(dtt, tied, tlDevice, tln, tExtRef, setting));
+        }
+    }
+
+    private void updateDaiValue(TDataTypeTemplates dtt, TIED tied, TLDevice tlDevice, TAnyLN tln, TExtRef tExtRef, TChannel setting) {
+        DO_DA_MAPPINGS.forEach(doNameAndDaName -> Optional.ofNullable(getNewDaiValue(doNameAndDaName.daName, LnId.from(tln).prefix(), tExtRef, setting))
+                .ifPresent(newDaiValue -> updateDaiVal(dtt, tied, tlDevice, tln, DoLinkedToDaFilter.from(doNameAndDaName.doName, doNameAndDaName.daName), newDaiValue)));
+    }
+
+    private String getNewDaiValue(String daName, String lnPrefix, TExtRef extRef, TChannel setting) {
+        return switch (daName) {
+            case DU_DA_NAME -> setting.isSetChannelShortLabel() ? setting.getChannelShortLabel(): null;
+            case SETVAL_DA_NAME -> {
+                if(LN_PREFIX_B.equals(lnPrefix) || LN_PREFIX_A.equals(lnPrefix)){
+                    yield setting.isSetChannelLevModQ() && !setting.getChannelLevModQ().equals(TChannelLevMod.NA) ? setting.getChannelLevModQ().value(): null;
+                } else {
+                    yield setting.isSetChannelLevMod() && !setting.getChannelLevMod().equals(TChannelLevMod.NA) ? setting.getChannelLevMod().value(): null;
+                }
+            }
+            case STVAL_DA_NAME -> ActiveStatus.ON.getValue();
+            case SETSRCREF_DA_NAME -> computeDaiValue(lnPrefix, extRef, setting.getDAName());
+            default -> throw new IllegalStateException("Unexpected value: " + setting.getDAName());
+        };
+    }
+
+    private void updateDaiVal(TDataTypeTemplates dtt, TIED tied, TLDevice tlDevice, TAnyLN tln, DoLinkedToDaFilter doLinkedToDaFilter, String newDaiValue) {
+        dataTypeTemplatesService.getFilteredDoLinkedToDa(dtt, tln.getLnType(), doLinkedToDaFilter)
+                .map(doLinkedToDa1 -> lnEditor.getDoLinkedToDaCompletedFromDAI(tied, tlDevice.getInst(), tln, doLinkedToDa1))
+                .findFirst()
+                .filter(doLinkedToDa1 -> {
+                    if (!doLinkedToDa1.isUpdatable()){
+                        errorHandler.add(SclReportItem.warning(tied.getName() + "/" + LDEVICE_LDSUIED + "/" + LnId.from(tln).lnClass() + "/DOI@name=\"" + doLinkedToDaFilter.doName() + "\"/DAI@name=\"" + doLinkedToDaFilter.daName() + "\"/Val", "The DAI cannot be updated"));
```

Comment:
Same comment for errorHandler as https://github.com/com-pas/compas-sct/pull/506#discussion_r2077258522

Reply:
Solved using `ThreadLocal`.